src/aosm/azext_aosm/common/registry.py

```diff
@@ -0,0 +1,497 @@
+                "acr",
+                "repository",
+                "show-tags",
+                "--name",
+                self.registry_name,
+                "--repository",
+                f"{repository}",
+            ]
+
+            version_list: List[str] = json.loads(
+                call_subprocess_raise_output(acr_get_versions_for_repository_cmd)
+            )
+
+            # The image name is the final part of the repository name
+            image = repository.split("/")[-1]
+            # The namespace is everything before the image name
+            namespace_list = repository.split("/")[:-1]
+            # If namespace is not an empty list then join it with a slash
+            if namespace_list:
+                namespace = "/".join(namespace_list) + "/"
+            else:
+                namespace = ""
+
+            for version in version_list:
+                image_dict[(image, version)] = (self, namespace)
+
+        return image_dict
+
+
+class ContainerRegistryHandler:
+    """
+    A class to handle all the registries and their images.
+    """
+
+    def __init__(self, image_sources):
+        self.image_sources = image_sources
+        self.registry_list = self._create_registry_list()
+        self.registry_for_image = self._get_registries_for_images()
+
+    def _create_registry_list(self) -> List[ContainerRegistry]:
+        """
+        Create a list of registry objects from the registries provided by the user.
+
+        :return: A list of registry objects
+        """
+
+        registries: Dict[str, ContainerRegistry] = {}
+        registry_list: List[ContainerRegistry] = []
+
+        logger.debug("Creating registry list from the provided registries")
+
+        for registry in self.image_sources:
+            registry = clean_registry_name(registry)
+
+            # If registry matches the pattern of myacr1.azurecr.io/**,
+            # then create a an instance of the ACR registry class
+            parts = registry.split("/", 1)
+            registry_name = parts[0]
+            registry_namespace = parts[1] if len(parts) > 1 else None
+
+            acr_match = re.match(ACR_REGISTRY_NAME_PATTERN, registry_name)
+
+            if registry_name not in registries:
+                if acr_match:
+                    registries[registry_name] = AzureContainerRegistry(registry_name)
+                else:
+                    registries[registry_name] = UniversalRegistry(registry_name)
+
+                registry_list.append(registries[registry_name])
+
+            if registry_namespace:
+                # Make sure that the namespace ends with a slash
+                if not registry_namespace.endswith("/"):
+                    registry_namespace += "/"
+                registries[registry_name].add_namespace(registry_namespace)
+            else:
+                registries[registry_name].add_namespace("")
+
+        return registry_list
+
+    def get_registry_list(self) -> List[ContainerRegistry]:
```

Comment:
I thought that interacting directly with the attribute of a class is generally less advisable than having a method that gets that attribute. But maybe that is only true for changing/updating attributes rather than just getting them. I will delete this method